source/Hardpoint.cpp

```diff
@@ -124,20 +124,20 @@ int Hardpoint::BurstRemaining() const
 
 
 
-// Perform one step (i.e. decrement the reload count).
-void Hardpoint::Step()
+// Perform many steps (i.e. decrement the reload count when the date changes).
+void Hardpoint::Step(int frames)
 {
 	if(!outfit)
 		return;
 	
 	wasFiring = isFiring;
 	if(reload > 0.)
-		--reload;
+		reload -= frames;
```

Comment:
okay. Most likely you will need to write something like `reload -= min(reload+1, frames);` to properly handle the weapons with reloads like 1.5 (= 40 shots per second).
these need to be allowed to have their current reload value slightly in the negative (but not below -1)